app/api/sections.js

```diff
@@ -4,21 +4,58 @@ const router = Router()
 const db = require('../models/index')
 const { logger } = require('../../lib/logger')
 const sectionService = require('../services/section_service')
-const { requireAuthentication, generateOTP } = require('../../lib/auth')
+const { requireAuthentication } = require('../../lib/auth')
 const { ValidationError, UniqueConstraintError } = require('sequelize')
 
-router.post('/:course_id/sections', requireAuthentication, async function (req, res) {
-    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
-    // make sure the user creating this section is the teacher for the course
-    const enrollment = await db.Enrollment.findOne({
+async function checkIfTeacher(userId, courseId) {
+    return await db.Enrollment.findOne({
         where: { 
-            userId: user.id,
-            courseId: parseInt(req.params['course_id']),
+            userId: userId,
+            courseId: courseId,
             role: 'teacher'
         }
     })
+}
+
+async function getSection(sectionId) {
+    return await db.Section.findOne({
+        where: { 
+            id: sectionId
+        }
+    })
+}
+
+async function getSectionsInCourse(courseId) {
+    return await db.Section.findAll({
+        where: {
+            courseId: courseId
+        }
+    })
+}
+
+async function getLecturesFromSection(sectionId) {
+    return await db.Lecture.findAll({
+        include: [{
+            model: db.LectureForSection,
+            include: [{
+                model: db.Section,
+                where: { id: sectionId },
+            }],
+            attributes: []  // don't return any attributes from join (only lecture fields)
```

Comment:
I think we might want some of the fields, no? LectureForSection includes publication information, along with some other fields. I might not understand the joining though. Let me know if I'm mistaken on how this is working